server/src/main/java/org/cloudfoundry/identity/uaa/provider/oauth/OauthIDPWrapperFactoryBean.java

```diff
@@ -166,6 +166,9 @@ protected void setCommonProperties(Map<String, Object> idpDefinitionMap, Abstrac
         if (idpDefinitionMap.get("clientAuthInBody") instanceof Boolean) {
             idpDefinition.setClientAuthInBody((boolean)idpDefinitionMap.get("clientAuthInBody"));
         }
+        if (idpDefinitionMap.get("performRpInitiatedLogout") instanceof Boolean) {
+            idpDefinition.setPerformRpInitiatedLogout((boolean)idpDefinitionMap.get("performRpInitiatedLogout"));
+        }
```

Comment:
I would like to have an `else` block here, throwing a type error exception. 

Couldn't a user set the value to some silly string like `tralse` in the configuration file, and the code just wouldn't say anything, just stay on the default value?

Reply:
The lack of error exception of malformed boolean config fields is an existing problem (if it is proven to be a problem, there could be some code elsewhere doing this erroring), so I would consider it out of scope of this PR, which I'm trying to scope minimally. 

But you are right. it could be a problem. I'll consider that for the "refactor PR" later. But the solution will need to apply to all the boolean configs (like `clientAuthInBody`).